pkg/middleware/instrumentation.go

```diff
@@ -0,0 +1,109 @@
+package middleware
+
+import (
+	"net/http"
+
+	"github.com/justinas/alice"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var metricsHandler = NewInstrumentMetricHandlerWithDefaultRegistry()
+
+func NewInstrumentMetricHandlerWithDefaultRegistry() http.Handler {
```

Comment:
Would be nice to have comments on each of the functions in this file as we do in others in this package